### PR TITLE
[Flow EVM] Avoid hex encoding/decoding for event fields of type bytes

### PR DESCRIPTION
**Pull request by ramtinms**
Most of the fields of data are RLP encoded and later the event will also be encoded by CCF. having another layer of hex encoding decoding impacts both the performance and compactness of events without any gain.

### CODE REVIEW

**Comment by ramtinms** [comment]
This is a breaking change but from the feedback we got from the community, the changes are mostly limited to the GW node.

**Review by janezpodhostnik — APPROVED** [review]
Looks good to me.
This requires an HCU.

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -46,8 +46,8 @@ contract EVM {
         index: UInt16,
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:46]: It's great to see making the type less "stringly typed" 👍 

Can't comment on the right line, but maybe the hash an also be changed in a similar way?

File: fvm/evm/stdlib/contract.cdc [hunk 2]
```diff
@@ -56,8 +56,8 @@ contract EVM {
         gasConsumed: UInt64,
         // if transaction was a deployment contains a newly deployed contract address
         contractAddress: String,
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:58]: While at it, maybe the contract address could also be improved and and changed to `Address`?

File: fvm/evm/types/utils.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+// CadenceUInt8ArrayValueToBytes converts a Cadence array of type UInt8 into a byte slice
+func CadenceUInt8ArrayValueToBytes(a cadence.Value) ([]byte, error) {
+	aa, ok := a.(cadence.Array)
+	if !ok {
+		return nil, fmt.Errorf("value is not an array")
+	}
+
+	// if array type is empty, continue
+	if aa.Type() != nil && !aa.Type().Equal(cadenceArrayTypeOfUInt8) {
```
Comment (turbolent) [inline fvm/evm/types/utils.go:30]: Maybe factor the repeated `aa.Type()` into a variable

**Review by turbolent — APPROVED** [review]
Nice!

Left suggestions for two related improvements. If wanted, happy to add them to this PR or put up a new PR for them

File: fvm/evm/stdlib/contract.cdc (hunk 2 above)
Comment (ramtinms) [reply fvm/evm/stdlib/contract.cdc:58]: I'll leave it for another PR, there was some discussion in the past by the community for checking deployment by comparing this value to empty, I need to check those comments first.

File: fvm/evm/types/utils.go (hunk 3 above)
Comment (ramtinms) [reply fvm/evm/types/utils.go]: updated

File: fvm/evm/stdlib/contract.cdc (hunk 1 above)
Comment (ramtinms) [reply fvm/evm/stdlib/contract.cdc:46]: Going to create a follow-up PR for it. good idea.

Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:46]: Nice! Have already some commits here if you want to grab them: https://github.com/onflow/flow-go/compare/ramtin/evm-avoid-hex-encoding-for-event-bytes...bastian/evm-improve-types

**Pull request merged**